src/test/descriptor_tests.cpp

```diff
@@ -137,16 +137,14 @@ void DoCheck(const std::string& prv, const std::string& pub, const std::string&
     std::unique_ptr<Descriptor> parse_pub;
     // Check that parsing succeeds.
     if (replace_apostrophe_with_h_in_prv) {
-        parse_priv = Parse(UseHInsteadOfApostrophe(prv), keys_priv, error);
-    } else {
-        parse_priv = Parse(prv, keys_priv, error);
+        prv = UseHInsteadOfApostrophe(prv);
     }
+    parse_priv = Parse(prv, keys_priv, error);
     BOOST_CHECK_MESSAGE(parse_priv, error);
     if (replace_apostrophe_with_h_in_pub) {
-        parse_pub = Parse(UseHInsteadOfApostrophe(pub), keys_pub, error);
-    } else {
-        parse_pub = Parse(pub, keys_pub, error);
+        pub = UseHInsteadOfApostrophe(pub);
     }
+    parse_pub = Parse(pub, keys_pub, error);
     BOOST_CHECK_MESSAGE(parse_pub, error);
```

Comment:
It's not clear to me why this and the change in `Check` are necessary?

Reply:
Could be a left-over from the squash. Will look into this once we decide whether or not to normalise private keys, because that might change things again: https://github.com/bitcoin/bitcoin/pull/26076#discussion_r1016668800

Reply:
Not a left-over; the test fails without it. The behaviour is quite different now that we echo the original descriptor rather than always put `'` in it.